### PR TITLE
Improve exception/error handling

### PR DESCRIPTION
**Pull request by no2chem**
## Overview

Description: This PR improves error and exception handling by throwing ```UnrecoverableCorfuError``` whenever an unexpected exception is caught, and sleeping using the ```Utils.sleepUninterruptibly``` helper, which throws a ```UnrecoverableCorfuInterruptedError``` when interrupted.

Why should this be merged: This PR addresses several cases where catching all ```Exception``` types masked errors which should have been passed on to the application. In particular, this type of handling was causing ```InterruptedException``` to not shutdown threads on application termination.

Related issue(s) (if applicable): #1033

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
 import org.corfudb.runtime.exceptions.TransactionAbortedException;
 import org.corfudb.runtime.exceptions.TrimmedException;
 import org.corfudb.runtime.exceptions.TrimmedUpcallException;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuError;
```
Comment (corfu-repo) [inline runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java]: MINOR Remove this unused import 'org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuError'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java [hunk 2]
```diff
@@ -24,6 +24,7 @@
...
 import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuInterruptedError;
```
Comment (corfu-repo) [inline runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java]: MINOR Remove this unused import 'org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuInterruptedError'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by corfu-repo** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **bd30941**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #1060 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/xexX5vrv4kdyiOzTpchvN3tJR2KwobXi)

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **f0162c4**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #1060 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/dqsobSmzaZtZR9Brxy2nM8eBpF2Plic0)

File: runtime/src/main/java/org/corfudb/runtime/view/ObjectsView.java [hunk 3]
```diff
@@ -178,37 +179,37 @@ public long TXEnd()
...
-                        snapshotTimestamp);
+                    snapshotTimestamp);
                 TransactionAbortedException tae = new TransactionAbortedException(txInfo,
-                        null, null, abortCause, e, context);
+                    null, null, AbortCause.NETWORK, e, context);
                 context.abortTransaction(tae);
                 throw tae;
+
+            } catch (Exception e) {
+               log.error("TXEnd: Unexpected exception", e);
```
[the remark this answers is not in the document]
Comment (no2chem) [reply runtime/src/main/java/org/corfudb/runtime/view/ObjectsView.java]: done

[the remark this answers is not in the document]
Comment (no2chem) [reply runtime/src/main/java/org/corfudb/runtime/view/ObjectsView.java]: done

File: runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java (hunk 1 above)
Comment (corfu-repo) [inline runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java]: MINOR Remove this unused import 'org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuError'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java (hunk 2 above)
Comment (corfu-repo) [inline runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java]: MINOR Remove this unused import 'org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuInterruptedError'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by corfu-repo** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**